### PR TITLE
Update requirements.txt

### PR DESCRIPTION
**Pull request by msmygit**
The current branch gives an error of 
```
Collecting cassio (from -r requirements.txt (line 2))
  Cloning https://github.com/cassioml/cassio (to revision SL-delete-on-md-search) to /tmp/pip-install-z1r3cxp6/cassio_405815a2631f441fb78988403c0076ae
  Running command git clone --filter=blob:none --quiet https://github.com/cassioml/cassio /tmp/pip-install-z1r3cxp6/cassio_405815a2631f441fb78988403c0076ae
  WARNING: Did not find branch or tag 'SL-delete-on-md-search', assuming revision or ref.
  Running command git checkout -q SL-delete-on-md-search
  error: pathspec 'SL-delete-on-md-search' did not match any file(s) known to git
  error: subprocess-exited-with-error
  
  × git checkout -q SL-delete-on-md-search did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
  
```
and also both `cassio` and `langchain` have gone past the required versions so it is safer to upgrade.

### CODE REVIEW

File: api/requirements.txt [hunk 1]
```diff
@@ -1,8 +1,8 @@
-# cassio>=0.1.1
-git+https://github.com/cassioml/cassio@SL-delete-on-md-search#egg=cassio
+cassio>=0.1.3
+# git+https://github.com/cassioml/cassio@SL-delete-on-md-search#egg=cassio
 
-# langchain>=0.0.274
-git+https://github.com/hemidactylus/langchain@updated-full-preview-remove-shims#egg=langchain&subdirectory=libs/langchain
+langchain>=0.0.309
```
Comment (hemidactylus) [inline api/requirements.txt]: We cannot yet move this demo to upstream (support for partitioning is not landed to LangChain yet), this must be reverted to that "remove-shims" branch in my fork, unfortunately.

File: api/requirements.txt [hunk 2]
```diff
@@ -1,8 +1,8 @@
-# cassio>=0.1.1
-git+https://github.com/cassioml/cassio@SL-delete-on-md-search#egg=cassio
+cassio>=0.1.3
```
Comment (hemidactylus) [inline api/requirements.txt:1]: Awesome, thank you for spotting that!

**Comment by msmygit** [comment]
There is also a related issue with Astra CLI/DevOps API where it doesn't accept a vector database region of Azure `westus3`. See https://github.com/datastax/astra-cli/issues/117 for more details.

**Comment by hemidactylus** [comment]
LGTM, thank you!

**Pull request merged**